src/Mvc/Mvc.Testing/src/WebApplicationFactory.cs

```diff
@@ -524,6 +524,7 @@ protected virtual void Dispose(bool disposing)
                 }
 
                 _server?.Dispose();
+                _host?.StopAsync().Wait();
```

Comment:
If we want to test that this is called, we could 
* Create the factory, request the `IapplicationLifetime` service from the host services (available through the factory).
* Register a callback on stop.
* Dispose the factory.
* Validate the callback was invoked.

Reply:
Very thanks @javiercn. I added the corresponding test as per your instructions, but instead of `IApplicationLifetime`, I used `IHostApplicationLifetime` as `IApplicationLifetime` is `obsolete`.

Reply:
Yes, I didn't remember exactly how it was named, so perfect!

Reply:
😀